bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -1032,27 +1018,31 @@ impl From<SdkShieldState> for ShieldState {
 
 #[derive(Clone, uniffi::Record)]
 pub struct EventTimelineItem {
-    is_local: bool,
+    /// Indicates that an event is remote.
     is_remote: bool,
-    event_or_transaction_id: EventOrTransactionId,
+    /// An event id, for events that have been remote-echoed.
+    event_id: Option<String>,
+    /// A transaction id, for events waiting in the send queue.
+    ///
+    /// Note: not set for remote-echoes.
+    transaction_id: Option<String>,
```

Comment:
This sounds a lot like `EventOrTransactionId` to me.

Reply:
We're already losing much static typing safety by using Strings everywhere, maybe we can afford two types to help distinguish one from the other 🥲 Also, one can't use an `EventOrTransactionId` in any API, it's either an event id OR a transaction id, so using any of those APIs would require passing the event|transaction id and pray your best luck/favorite god that it doesn't fail because it wasn't the expected type.